Cargo.toml

```diff
@@ -52,7 +52,8 @@ authors = ["Aya Contributors"]
 license = "MIT OR Apache-2.0"
 repository = "https://github.com/aya-rs/aya"
 homepage = "https://aya-rs.dev"
-edition = "2021"
+rust-version = "1.85.0"
```

Comment:
I know we had talked about setting an MSRV in the past, but are there specific features that drive the need for this (OwnedFd was one that came to mind)?
Adding the rationale for the version requirement in the README would be useful so we can direct users there if they have questions.

Reply:
Yes I know that. But this is a breaking change for users, so please update `BREAKING_CHANGES.md` and elaborate a little more on what we gain by moving to the 2024 edition there.

Reply:
Done.